### PR TITLE
Replace all asserts with chai.should

### PR DESCRIPTION
**Pull request by nventuro**
Fixes part of #1091.

### CODE REVIEW

File: test/LimitBalance.test.js [hunk 1]
```diff
@@ -3,6 +3,12 @@ const { ethGetBalance } = require('./helpers/web3');
 
 const LimitBalanceMock = artifacts.require('LimitBalanceMock');
 
+const BigNumber = web3.BigNumber;
```
Comment (shrugs) [inline test/LimitBalance.test.js]: I don't like doing this pattern if we never use BigNumber by itself again. can we use

```
require('chai')
  .use(require('chai-bignumber')(web3.BigNumber))
  .should();
```

Comment (nventuro) [reply test/LimitBalance.test.js]: Some test files do use it though (but I'd have to check and see if it's necessary). Maybe we could add some sort of global helper everyone requires, that initializes chai and exposes both bignumber and should.

Comment (shrugs) [reply test/LimitBalance.test.js]: actually yeah I was wondering about that; just looked it up and we can do this https://github.com/chaijs/chai/pull/868#issuecomment-261361587 to auto-register should, but we'd have to add a custom one to do bignumber.

Comment (nventuro) [reply test/LimitBalance.test.js]: I think just having that one `require` (and making our setup not so magic) is fine for now.

Comment (shrugs) [reply test/LimitBalance.test.js]: 👍

Comment (nventuro) [reply test/LimitBalance.test.js]: I'll add that once this is merged then, to keep the PRs tidy.

File: test/ownership/HasNoTokens.test.js [hunk 2]
```diff
@@ -25,10 +32,12 @@ contract('HasNoTokens', function ([_, owner, initialAccount, anyone]) {
...
     const ownerStartBalance = await token.balanceOf(owner);
     await hasNoTokens.reclaimToken(token.address, { from: owner });
+
     const ownerFinalBalance = await token.balanceOf(owner);
     const finalBalance = await token.balanceOf(hasNoTokens.address);
-    assert.equal(finalBalance, 0);
-    assert.equal(ownerFinalBalance - ownerStartBalance, 10);
+
+    finalBalance.should.be.bignumber.equal(0);
+    (ownerFinalBalance.sub(ownerStartBalance)).should.be.bignumber.equal(10);
```
Comment (frangio) [inline test/ownership/HasNoTokens.test.js]: Unnecessary parentheses here.

File: test/payment/SplitPayment.test.js [hunk 3]
```diff
@@ -37,58 +37,58 @@ contract('SplitPayment', function ([_, owner, payee1, payee2, payee3, nonpayee1,
...
-      initBalance.should.be.bignumber.equal(amount);
+      initBalance.should.be.bignumber.eq(amount);
 
       // distribute to payees
       const initAmount1 = await ethGetBalance(payee1);
       await this.contract.claim({ from: payee1 });
-      const profit1 = await ethGetBalance(payee1) - initAmount1;
-      assert(Math.abs(profit1 - web3.toWei(0.20, 'ether')) < 1e16);
+      const profit1 = (await ethGetBalance(payee1)).sub(initAmount1);
+      profit1.sub(web3.toWei(0.20, 'ether')).abs().should.be.bignumber.lt(1e16);
```
Comment (frangio) [inline test/payment/SplitPayment.test.js]: This is such a weird test.

File: test/token/ERC20/StandardToken.test.js [hunk 4]
```diff
@@ -417,7 +423,7 @@ contract('StandardToken', function ([_, owner, recipient, anotherAccount]) {
             await this.token.increaseApproval(spender, amount, { from: owner });
 
             const allowance = await this.token.allowance(owner, spender);
-            assert.equal(allowance, amount + 1);
+            allowance.should.be.bignumber.eq(amount + 1);
```
Comment (frangio) [inline test/token/ERC20/StandardToken.test.js]: Is `amount` here a BigNumber instance? We should be using BigNumber addition.

File: test/token/ERC20/StandardToken.test.js [hunk 5]
```diff
@@ -453,7 +459,7 @@ contract('StandardToken', function ([_, owner, recipient, anotherAccount]) {
...
```
Comment (frangio) [inline test/token/ERC20/StandardToken.test.js]: Same here.

File: test/payment/SplitPayment.test.js (hunk 3 above)
Comment (nventuro) [reply test/payment/SplitPayment.test.js]: Yup, we need to revise all tests at some point, some have poor coverage/are a bit wonky.

File: test/token/ERC20/StandardToken.test.js (hunk 4 above)
Comment (nventuro) [reply test/token/ERC20/StandardToken.test.js]: It's not, though it probably should be. Spawned https://github.com/OpenZeppelin/openzeppelin-solidity/issues/1189

**Pull request merged**